builtin/providers/aws/resource_aws_api_gateway_method.go

```diff
@@ -0,0 +1,221 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayMethod() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayMethodCreate,
+		Read:   resourceAwsApiGatewayMethodRead,
+		Update: resourceAwsApiGatewayMethodUpdate,
+		Delete: resourceAwsApiGatewayMethodDelete,
+
+		Schema: map[string]*schema.Schema{
+			"api_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"http_method": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"authorization": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"api_key_required": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"request_models": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayMethodCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	models := make(map[string]string)
+	for k, v := range d.Get("request_models").(map[string]interface{}) {
+		models[k] = v.(string)
+	}
+
+	parameters := make(map[string]bool)
+	if parameterData, ok := d.GetOk("request_parameters"); ok {
+		params := parameterData.(*schema.Set).List()
+		for k := range params {
+			parameters[params[k].(string)] = true
+		}
+	}
+
+	_, err := conn.PutMethod(&apigateway.PutMethodInput{
+		AuthorizationType: aws.String(d.Get("authorization").(string)),
+		HttpMethod:        aws.String(d.Get("http_method").(string)),
+		ResourceId:        aws.String(d.Get("resource_id").(string)),
+		RestApiId:         aws.String(d.Get("api_id").(string)),
+		RequestModels:     aws.StringMap(models),
+		RequestParameters: nil,
```

Comment:
Why are we forcing this to be `nil` and not letting the user to specify this via schema/DSL?